src/primitives.es6

```diff
@@ -0,0 +1,339 @@
+    super('string', content, meta, attributes);
+  }
+
+  get length() {
+    return this.content.length;
+  }
+}
+
+export class NumberType extends ElementType {
+  constructor(content, meta, attributes) {
+    super('number', content, meta, attributes);
+  }
+}
+
+export class BooleanType extends ElementType {
+  constructor(content, meta, attributes) {
+    super('boolean', content, meta, attributes);
+  }
+}
+
+class Collection extends ElementType {
+  get length() {
+    return this.content.length;
+  }
+
+  get [Symbol.iterator]() {
+    return this.content[Symbol.iterator];
+  }
+
+  toValue() {
+    return this.content.map((el) => el.toValue());
+  }
+
+  toRefract() {
+    return super.toRefract({
+      content: this.content.map((el) => el.toRefract())
+    });
+  }
+
+  toCompactRefract() {
+    const attributes = this.convertAttributesToRefract('toCompactRefract');
+    const compactDoms = this.content.map((el) =>
+      el.toCompactRefract());
+    return [this.element, this.meta, attributes, compactDoms];
+  }
+
+  fromRefract(dom) {
+    this.element = dom.element;
+    this.meta = dom.meta;
+    this.attributes = dom.attributes;
+    this.content = (dom.content || []).map((content) =>
+      convertFromRefract(content));
+
+    this.convertAttributesToElements(convertFromRefract);
+
+    return this;
+  }
+
+  fromCompactRefract(tuple) {
+    this.element = tuple[0];
+    this.meta = tuple[1];
+    this.attributes = tuple[2];
+    this.content = (tuple[3] || []).map((content) =>
+      convertFromCompactRefract(content));
+
+    this.convertAttributesToElements(convertFromCompactRefract);
+
+    return this;
+  }
+
+  get(index) {
+    return index === undefined ? this : this.content[index];
+  }
+
+  set(index, value) {
+    this.content[index] = convertToType(value);
+    return this;
+  }
+
+  map(cb) {
+    return this.content.map(cb);
+  }
+
+  filter(condition) {
+    const newArray = new Collection();
+    newArray.content = this.content.filter(condition);
+    return newArray;
+  }
+
+  forEach(cb) {
+    this.content.forEach(cb);
+  }
+
+  push(value) {
+    this.content.push(convertToType(value));
+    return this;
+  }
+
+  add(value) {
+    this.push(value);
+  }
+
+  findElements(condition, options={}) {
+    const recursive = options.recursive === undefined ? false : options.recursive;
+    const results = options.results === undefined ? [] : options.results;
+
+    this.content.forEach((el) => {
+      // We use duck-typing here to support any registered class that
+      // may contain other elements.
+      if (recursive && (el.findElements !== undefined)) {
+          el.findElements(condition, {results, recursive});
+      }
+      if (condition(el)) {
+        results.push(el);
+      }
+    });
+    return results;
+  }
+
+  find(condition, options) {
+    const newArray = new Collection();
+    newArray.content = this.findElements(condition, options);
+    return newArray;
+  }
```

Comment:
I noticed in the reveal today, you used `findElements` instead of `find`. Are you wanting to move away from `find` or did `findElements` work better?

Just as a note, if you're wanting to move away from it, the README still shows `find` examples.